iXBRLViewerPlugin/viewer/src/js/viewer.js

```diff
@@ -110,6 +110,40 @@ Viewer.prototype._addDocumentSetTabs = function() {
     }
 }
 
+/*
+ * Rewrite hyperlinks in the iXBRL.
+ *
+ * Relative links to other files in the same document set are handled by
+ * JavaScript to switch tabs within the viewer
+ *
+ * All other links are forced to open in a new tab
+ *
+ */
+Viewer.prototype._updateLink = function(n) {
+    const url = $(n).attr("href");
+    if (url !== undefined) {
+        const [file, fragment] = url.split('#', 2);
+        const docSetFiles = this._report.documentSetFiles();
+        if (this._report.isDocumentSet() && !url.includes('/') && docSetFiles.includes(file)) {
+            $(n).click((e) => { 
+                const index = docSetFiles.indexOf(file);
+                this.selectDocument(index); 
+                if (fragment !== undefined && fragment != "") {
+                    const f = $.escapeSelector(fragment);
+                    const ee = this._iframes.eq(index).contents().find('#' + f + ', a[name="' + f + '"]');
+                    if (ee.length > 0) {
+                        this.showElement(ee.eq(0));
+                    }
+                }
+                e.preventDefault(); 
+            });
+        }
+        else {
+            $(n).attr("target", "_blank");
```

Comment:
Perhaps a small comment to explain why we're doing this?
> Load external sites in a new tab to avoid loading them in the viewer iframe.

A new tab seems reasonable since users probably want to keep their viewer state, but I suppose `_top` might also be a reasonable solution.

Reply:
Done.  

Agree on the choice between `_blank` and `_top`.  Now that we're handling intra-documentSet links within the viewer, it seems somehow worse that other links might replace the viewer, so I have a mild preference for `_blank`